packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -138,27 +169,80 @@ export class TableLayoutManager<TData extends TableRecord> {
         let availableSpace = 0;
         if (requestedResizeAmount > 0) {
             // size right
-            return requestedResizeAmount;
+            return this.hasResizableColumnToLeft(this.leftColumnIndex!)
+                ? requestedResizeAmount
+                : 0;
         }
 
         // size left
-        let currentIndex = this.leftColumnIndex!;
-        while (currentIndex >= 0) {
-            const columnInitialWidths = this.initialColumnWidths[currentIndex]!;
-            availableSpace
-                += columnInitialWidths.initialPixelWidth
-                - columnInitialWidths.minPixelWidth;
-            currentIndex -= 1;
+        if (!this.hasResizableColumnToRight(this.rightColumnIndex!)) {
+            return 0;
+        }
+
+        for (let i = this.leftColumnIndex!; i >= 0; i--) {
+            const columnInitialWidths = this.initialColumnWidths[i]!;
+            if (!columnInitialWidths.resizingDisabled) {
+                availableSpace
+                    += columnInitialWidths.initialPixelWidth
+                    - columnInitialWidths.minPixelWidth;
+            }
         }
         return Math.max(requestedResizeAmount, -availableSpace);
     }
 
+    /**
+     * Gets the index of the first resizable column starting with
+     * `columnIndex` and moving to the left. If no resizable column
+     * is found, returns -1.
+     */
+    private getFirstLeftResizableColumnIndex(
+        columnIndex: number,
+        visibleColumns = this.visibleColumns
+    ): number {
+        for (let i = columnIndex; i >= 0; i--) {
+            const column = visibleColumns[i];
+            if (!column) {
+                return -1;
+            }
+            if (!column.columnInternals.resizingDisabled) {
```

Comment:
I didn't try and trace the logic too deeply but one thing that is not clear to me in this class is when we access the cached state vs the live state of these properties. My initial thought is that after caching all these code paths should generally access the cache value instead of the current columnInternals value. If that's the case can we double check the columnIInternals accesses in the file? If not curious how to make it clear for future authors.

Reply:
I've updated the code a bit to hopefully make things clearer. Now, the cached state only includes the initial width values, which are not available in the column/columnInternals itself. Any other state, such as minimum width or resizing disabled is read from the column.

Reply:
That helps! 👍